agilent_infiniium_scope.py

```diff
@@ -31,360 +31,389 @@
     def get_trace(self, c, channel, start=1, stop=10000):
         """Get a trace from the scope.
         OUTPUT - (array voltage in volts, array time in seconds)
         """
-        #raise Exception('Doesnt work yet. Please fix lines defining traceVolts and time.')
-##        DATA ENCODINGS
-##        RIB - signed, MSB first
-##        RPB - unsigned, MSB first
-##        SRI - signed, LSB first
-##        SRP - unsigned, LSB first
-        wordLength = 2 #Hardcoding to set data transer word length to 2 bytes
+        # raise Exception('Doesnt work yet. Please fix lines defining
+        # trace_volts and time.')
+        # DATA ENCODINGS
+        # RIB - signed, MSB first
+        # RPB - unsigned, MSB first
+        # SRI - signed, LSB first
+        # SRP - unsigned, LSB first
+        word_length = 2  # Hardcoding to set data transer word length to 2 bytes
         
         dev = self.selectedDevice(c)
-        yield dev.write('WAV:SOUR CHAN%d' %channel)
+        yield dev.write('WAV:SOUR CHAN{}'.format(channel))
 
-        #Read data MSB first
+        # Read data MSB first
         yield dev.write('WAV:BYT MSBF')
-        #Set 2 bytes per point
+        # Set 2 bytes per point
         yield dev.write('WAV:FORM WORD')
-        #Starting and stopping point
-        #Transfer waveform preamble
+        # Starting and stopping point
+        # Transfer waveform preamble
         preamble = yield dev.query('WAV:PRE?')
-        #Transfer waveform data
-        binary = yield dev.query('WAV:DATA?')
-        #Parse waveform preamble
-        preambleDict = _parsePreamble(preamble)
-        print preambleDict
-        #Parse binary
-        trace = _parseBinaryData(binary, wordLength = wordLength) * 1.0e3
-        #Convert from binary to volts
-        traceVolts = ((trace*float(preambleDict['yStep'])+float(preambleDict['yOrigin'])))
-        numPoints = int(preambleDict['numPoints'])
-        time = numpy.linspace(float(preambleDict['xFirst']),
-                              (numPoints-1) * float(preambleDict['xStep'])+float(preambleDict['xFirst']),
-                              numPoints)
-
-        returnValue((time*U.ns*1e9, traceVolts*U.V))
+        # Transfer waveform data
+        p = dev._packet().write('WAV:DATA?').read_raw()
+        result = yield p.send()
+        binary = result['read_raw']
+        binary = binary[:-1]
+        # Parse waveform preamble
+        preample_dict = _parsePreamble(preamble)
+        # Parse binary
+        trace = _parseBinaryData(binary, word_length=word_length)
+        # Convert from binary to volts
+
+        y_step = float(preample_dict['yStep'])
+        origin = float(preample_dict['yOrigin'])
+        trace_volts = (trace * y_step) + origin
+
+        num_points = int(preample_dict['numPoints'])
+        x_step = float(preample_dict['xStep'])
+        first = float(preample_dict['xFirst'])
+        time = numpy.linspace(first, first + (num_points-1) * x_step, num_points)
```

Comment:
I would call this `time_s` since it is in units of seconds. Then:
```python
time_ns = time_s * 1e9
returnValue((time_ns*U.ns, trace_volts*U.V))
```
(time_s*U.s).inUnitsOf